draco/analysis/ringmapmaker.py

```diff
@@ -821,93 +856,60 @@ def B(phi, u, sigma):
             """Azimuthal beam transfer function."""
             return np.exp(2.0j * np.pi * u * np.sin(phi)) * A(phi, sigma)
 
-        # Deteremine the RA axis from the maximum m-mode in the hybrid visibilities
+        # Determine the RA axis from the maximum m-mode in the hybrid visibilities
         mmax = hybrid_vis_m.mmax
         is_odd = np.any(hybrid_vis_m.vis[mmax, 1] != 0.0)
+        is_odd = hybrid_vis_m.comm.allreduce(is_odd)
```

Comment:
Should we add this `allreduce` to the other places where we determine if it's an even or odd number of RAs?  This is done on line 550 in the `DeconvolveHybridMBase`, and then also in the `_unpack_marray` used by `MModeInverseTransform`.

I guess a scenario where those instances would fail is when there is an odd number of RAs and some missing frequencies that are entirely zero.  If a rank handles only those missing frequencies then this expression would evaluate False for that rank only.

Reply:
Good point. I think the thing to do here is to keep this as a property on `MContainer`, it can be set by the MModeTransform class as it knows what the original set of RAs are, and then just be looked up by these other tasks.